### PR TITLE
add support for webpack4

### LINKED ISSUE #111: Support webpack 4

### PR DESCRIPTION
**Pull request by andriijas**
Please review carefully, Im not sure the PR is complete. I have no deep knowledge of webpack plugin internals. 

I tested it in a small reference app and got a manifest generated.

Feedback appreciated.

Fixes #111

### CODE REVIEW

**Comment by mastilver** [comment]
Thank you very much for that! :+1: 

Can you edit the `.travis.yml` so test are running for v4

**Comment by andriijas** [comment]
I updated it but I think we will have to wait until webpack-contrib/extract-text-webpack-plugin#707 to improve webpack 4 testing strategies.

**Comment by mastilver** [comment]
Cool test are passing, I will merge that this week-end (remind me if I don't... ;) )

**Comment by weaverryan** [comment]
@mastilver Ping also on making a stable 2.0 release :) :)

**Comment by ganapativs** [comment]
@mastilver When can we expect this to get merged? 😬

**Comment by irudoy** [comment]
Also, https://github.com/webpack-contrib/extract-text-webpack-plugin/pull/707 has been merged 🎉

**Comment by andriijas** [comment]
For me its totally OK to upgade all dependencies, cut a 2.0 release of the plugin only working with webpack 4 and clean up old plugin API but I guess most people want to have a transition period.

File: lib/plugin.js [hunk 1]
```diff
@@ -61,7 +59,7 @@ ManifestPlugin.prototype.apply = function(compiler) {
           path: path,
           chunk: chunk,
           name: name,
-          isInitial: chunk.isInitial ? chunk.isInitial() : chunk.initial,
+          isInitial: chunk.isInitial ? chunk.isInitial() : chunk.isOnlyInitial(),
```
Comment (wonism) [inline lib/plugin.js]: `isInitial` is only in `v3`.
So, I think `l62` will throw error in `v1/v2`. and it should be like this.
```
isInitial: chunk. isOnlyInitial ? chunk. isOnlyInitial() : (chunk.isInitial ? chunk.isInitial() : chunk.initial),
```

**Comment by mastilver** [comment]
@andriijas Thank you for the great work, let me know if you want ot be more involved into this project! :)

@all sorry for the delay, I didn't have much time. My boss just granted me permission to work on open source for the day, so I'm going to make the most out of it :)

**Comment by andriijas** [comment]
@mastilver  Thank you for letting me, I honestly dont have any deep knowledge  of webpack I tried my best using the migration guide: https://medium.com/webpack/webpack-4-migration-guide-for-plugins-loaders-20a79b927202

Im still not sure how callbacks changed with the new plugin API

**Comment by mastilver** [comment]
hum... weird, the build is now failing... maybe a dependency changed?

**Comment by mastilver** [comment]
Found where it's coming from, a test :)

**Comment by andriijas** [comment]
the test suite seems to run on webpack 3.x and maybe it should for a while. I read somewhere webpack 5 will be out in roughly  6 months, by then we can deprecate old plugin format.

Maybe we should use tapAsync though if we want that callback support   ref: https://github.com/facebook/create-react-app/pull/4077#discussion_r170622660

**Comment by mastilver** [comment]
No, I'm talking about our unit tests and also I think we are missing: `webpack-manifest-plugin-after-emit` now on webpack 4

**Comment by andriijas** [comment]
@mastilver check new PR

**Comment by mastilver** [comment]
`v2` is now available :tada:

Thank you @andriijas for your work, and all for your patience

**Comment by andriijas** [comment]
no worries, just been learning while helping.   thanks finishing up!

**Pull request merged**